www/scripts/productlist/ProductSettingsView.js

```diff
@@ -0,0 +1,717 @@
+              errors.push(record);
+            }
+          });
+
+          if (errors.length > 0) {
+            var text = "<ul>";
+            errors.forEach(function(record) {
+              var permissionName = util.enumValueToKey(AUTH_OBJECTS.Permission,
+                                                       record.permission);
+              text += '<li><strong>' + (record.action === 'ADD'
+                                        ? "Add" : "Remove") +
+                      '</strong> permission <strong>' + permissionName +
+                      '</strong> of ' + (record.isGroup ? "group" : "user") +
+                      ' <strong>' + record.name + '</strong>.</li>\n';
+            });
+            text += '</ul>';
+
+
+            that._showMessageBox(
+              'error',
+              "Some permission changes could not be saved",
+              text);
+          } else {
+            that._showMessageBox(
+              'success',
+              "Permission changes saved successfully!",
+              "");
+          }
+        }
+      });
+    },
+
+    _showMessageBox : function (mode, header, text) {
+      domClass.toggle(this._mbox.domNode, "mbox-error", mode === 'error');
+      domClass.toggle(this._mbox.domNode, "mbox-success", mode === 'success');
+      this._mbox.show(header, text);
+    },
+
+    populatePermissions : function(productID) {
+      this.permissionView.populatePermissions('PRODUCT', {
+        productID : productID
+      });
+    },
+
+    postCreate : function () {
+      this.inherited(arguments);
+      this.addChild(this._mbox);
+      this._mbox.hide();
+      this.addChild(this.permissionView);
+      this.addChild(this._btnSave);
+    }
+  });
+
+  return declare(Dialog, {
+    constructor : function () {
+      this._tabWrapper = new ContentPane({
+        style : 'width: 650px; height: 450px'
+      });
+
+      this._tabView = new TabContainer();
+
+      this._metadataPane = new ProductMetadataPane({
+        title    : "Product settings",
+        closable : false
+      });
+
+      this._permissionPane = new ProductPermissionsPane({
+        title    : "Permissions",
+        closable : false
+      });
+    },
+
+    postCreate : function () {
+      this.inherited(arguments);
+
+      this._tabView.addChild(this._metadataPane);
+      this._tabView.addChild(this._permissionPane);
+
+      this._tabWrapper.addChild(this._tabView);
+      this.addChild(this._tabWrapper);
+    },
+
+    onHide : function () {
+      this._metadataPane.resetForm();
+    },
+
+    setMode : function (adminLevel, mode, productID, successCallback) {
+      this._tabView.selectChild(this._metadataPane);
+
+      this._metadataPane.setAdminLevel(adminLevel);
+      this._metadataPane.set('successCallback', successCallback);
+
+      if (mode === 'add') {
+        this.set('title', "Add new product");
+        this._metadataPane.set('settingsMode', 'add');
+        this._permissionPane.set('disabled', true);
```

Comment:
It would be better to hide the `Permissions` tab instead of disabling it. It's annoying that I cannot go to that tab when I create a new product.